server/service/src/sync/translations/special/item_merge.rs

```diff
@@ -0,0 +1,157 @@
+use repository::{
+    ItemLinkRow, ItemLinkRowRepository, StorageConnection, SyncBufferAction, SyncBufferRow,
+};
+
+use serde::Deserialize;
+
+use crate::sync::translations::{
+    IntegrationRecords, LegacyTableName, PullDependency, PullUpsertRecord, SyncTranslation,
+};
+
+#[allow(non_snake_case)]
+#[derive(Deserialize)]
+pub struct ItemMergeMessage {
+    pub mergeIdToKeep: String,
+    pub mergeIdToDelete: String,
+}
+
+// Conceptually this isn't a translation, so the abstraction should probably be changed or this doesn't belong here
+pub(crate) struct ItemMergeTranslation {}
+impl SyncTranslation for ItemMergeTranslation {
+    fn pull_dependencies(&self) -> PullDependency {
+        PullDependency {
+            table: LegacyTableName::ITEM,
+            dependencies: vec![],
+        }
+    }
+
+    fn try_translate_pull_merge(
+        &self,
+        connection: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        if sync_record.table_name != LegacyTableName::ITEM
+            || sync_record.action != SyncBufferAction::Merge
+        {
+            return Ok(None);
+        }
+
+        let data = serde_json::from_str::<ItemMergeMessage>(&sync_record.data)?;
+
+        let item_link_repo = ItemLinkRowRepository::new(connection);
+        // let mut item_links = vec![];
+        let item_links = item_link_repo.find_many_by_item_id(&data.mergeIdToDelete)?;
+        if item_links.len() == 0 {
+            return Ok(None);
+        }
+        let indirect_link = item_link_repo.find_one_by_id(&data.mergeIdToKeep)?.unwrap();
+
+        let upsert_records: Vec<PullUpsertRecord> = item_links
+            .into_iter()
+            .map(|ItemLinkRow { id, .. }| {
+                PullUpsertRecord::ItemLink(ItemLinkRow {
+                    id,
+                    item_id: indirect_link.item_id.clone(),
+                })
+            })
+            .collect();
+
+        Ok(Some(IntegrationRecords::from_upserts(upsert_records)))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::sync::synchroniser::integrate_and_translate_sync_buffer;
+
+    use super::*;
+    use repository::{
+        mock::MockDataInserts, test_db::setup_all, ItemLinkRowRepository, SyncBufferAction,
+        SyncBufferRow, SyncBufferRowRepository,
+    };
+
+    #[actix_rt::test]
+    async fn test_item_merge() {
+        util::init_logger(util::LogLevel::Info);
+
+        let mut sync_records = vec![
+            SyncBufferRow {
+                record_id: "item_b_merge".to_string(),
+                table_name: LegacyTableName::ITEM.to_string(),
+                action: SyncBufferAction::Merge,
+                data: r#"{
+                        "mergeIdToKeep": "item_b",
```

Comment:
Is this the actual merge record ? I did a quick search for "mergeIdToKeep" in mSupply, and found "mergeIDtokeep"

Reply:
Yep in `develop`: https://github.com/msupply-foundation/msupply/blob/f1f175afefc05ce0a9d3bbf0b03b68afe59d9351/Project/Sources/Methods/syncOutToSyncV5.4dm#L29

In msupply master it is a bit different as `$obSyncRecord.mergeIdToKeep` rather than `$obSyncRecord.data.mergedIdToKeep`. Further back in history of the file it was `mergeIDtoKeep` though: https://github.com/msupply-foundation/msupply/blame/5a763ac929f1ec0090db805e901c5b1506e00fba/Project/Sources/Methods/syncOutToSyncV5.4dm#L105-L107

Do you have latest pulled from origin?

Reply:
Do we need to add compatibility restrictions ? What would happen if we try previous version merge records with this omSupply version ?

Reply:
Should have said - older versions of merge records were never sent to omSupply anyway. In the future if we did change the shape of `data` then yea I think we would need to bump the compatibility restrictions.